server/models/Availability.js

```diff
@@ -0,0 +1,52 @@
+const mongoose = require("mongoose");
+
+const dateOptions = {
+    startTime: {
+      type: Date, 
+      required: [
+        { validator: isDayActive, msg: 'A start time is required if this day is active!'},
+        { validator: isValidTime, msg: 'The start time cannot be after the end time'}
+      ]
+    }, 
+    endTime: {
+      type: Date,
+      required: [
+        { validator: isDayActive, msg: 'A end time is required if this day is active!'},
+        { validator: isValidTime, msg: 'The end time cannot be before the start time'}
+      ]
+    },
+    active: {
+      type: Boolean,
+      default: false, 
+    }
+}
+
+function isDayActive(){
+  return this.active === true
+}
+
+function isValidTime(){
+  return this.startTime > this.endTime
```

Comment:
```suggestion
  return this.startTime < this.endTime
```
Endtime should be greater than the start time